src/dataframe/dataframe.jl

```diff
@@ -16,6 +16,8 @@ DataFrame(columns::NTuple{N,AbstractVector}, names::NTuple{N,Symbol};
 DataFrame(columns::Matrix, names::Vector{Symbol}; makeunique::Bool=false)
 DataFrame(kwargs...)
 DataFrame(pairs::Pair{Symbol}...; makeunique::Bool=false, copycols::Bool=true)
+DataFrame(pairs::AbstractVector{Pair{Symbol, <:AbstractVector}}; copycols::Bool=true)
```

Comment:
This isn't completely correct IIUC: we also accept `Vector{Any}` as long as the entries are of the right type.

Reply:
But that's still checked in the fallback constructor, right?

Reply:
> This isn't completely correct IIUC

You are right, but I did not know how to express this better (that is why in the commit message I have said this). Since this PR is merged if you can propose a better docstring please advise.

A precise specification would be that "we accept `AbstractVector` of any type as long as all its entries are subtypes of `Pair{Symbol, <:AbstractVector}`" but I was not clear how to express it in a concise way.

Reply:
I guess the only correct way to document this is to use `::AbstractVector` here, and give more details in the text.

Reply:
OK - I will fix it in #1790